src/main/scala/com/box/castle/batch/CastleMessageBatch.scala

```diff
@@ -90,6 +90,19 @@ object CastleMessageBatch {
     new CastleMessageBatch(toIndexedSeq(messageSetCopy), messageSet.sizeInBytes)
   }
 
+  /**
+    * This method constructs a new CastleMessageBatch from a vector of consecutive CastleMessageBatches.
+    * NOTE: In this case we don't need to do a deep copy since that was already done when the CastleMessageBatches
+    * in the vector were created.
+    * @param messageBatches
+    * @return
+    */
+  def apply(messageBatches: Vector[CastleMessageBatch]): CastleMessageBatch = {
```

Comment:
this needs a runtime check to make sure that all the offsets in the vector are contiguous, or there will be a chance to miss some messages

Reply:
Done